framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -275,6 +275,34 @@
     protected AnnotationClassLoader loader;
 
     /**
+     * Should results be cached?
+     * This means that ATM.deepCopy() will be called.
+     * ATM.deepCopy() use to (and perhaps still does) side effect the ATM being copied.
+     * So setting this to false is not equivalent to setting shouldReadCache to false.
+     * */
+    public boolean shouldCache;
+
+    /** Should the cached result be used, or should it be freshly computed? */
+    public boolean shouldReadCache;
+
+    /** Size of LRU cache. */
```

Comment:
Relation to getCacheSize should be documented. This is the default size, not the actual size.